File: sdk/java/src/main/java/com/anyvali/parse/CoercionConfig.java
package com.anyvali.parse;

/**
 * Configuration for coercion behavior.
 */
public record CoercionConfig(
        boolean toInt,
        boolean toNumber,
        boolean toBool,
        boolean trim,
        boolean lower,
        boolean upper
) {
    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        private boolean toInt;
        private boolean toNumber;
        private boolean toBool;
        private boolean trim;
        private boolean lower;
        private boolean upper;

        public Builder toInt(boolean v) { this.toInt = v; return this; }
        public Builder toNumber(boolean v) { this.toNumber = v; return this; }
        public Builder toBool(boolean v) { this.toBool = v; return this; }
        public Builder trim(boolean v) { this.trim = v; return this; }
        public Builder lower(boolean v) { this.lower = v; return this; }
        public Builder upper(boolean v) { this.upper = v; return this; }

        public CoercionConfig build() {
            return new CoercionConfig(toInt, toNumber, toBool, trim, lower, upper);
        }
    }
}

File: sdk/kotlin/src/main/kotlin/com/anyvali/parse/CoercionConfig.kt
package com.anyvali.parse

data class CoercionConfig(
    val type: String // e.g. "string->int", "string->number", "string->bool", "trim", "lower", "upper"
) {
    companion object {
        val STRING_TO_INT = CoercionConfig("string->int")
        val STRING_TO_NUMBER = CoercionConfig("string->number")
        val STRING_TO_BOOL = CoercionConfig("string->bool")
        val TRIM = CoercionConfig("trim")
        val LOWER = CoercionConfig("lower")
        val UPPER = CoercionConfig("upper")
    }
}
